lib/js/src/rpc/messages/RegisterAppInterface.js

```diff
@@ -0,0 +1,381 @@
+                appID = fullAppID;
+            } else {
+                appID = fullAppID.replace("-", "").substring(0, RegisterAppInterface.APP_ID_MAX_LENGTH);
+            }
+            this._setAppId(appID);
+        } else {
+            setParameters(RegisterAppInterface.KEY_FULL_APP_ID, null);
+        }
+
+        return this;
+    }
+
+    /**
+    * @return {String} the app id
+    */
+    getFullAppId() {
+        return this.getParameter(RegisterAppInterface.KEY_FULL_APP_ID);
+    }
+
+
+    /**
+     * @param {AppInfo} appInfo
+     * @return {RegisterAppInterface}
+     */
+    setAppInfo(appInfo) {
+        this.validateType(AppInfo, appInfo);
+
+        this.setParameter(RegisterAppInterface.KEY_APP_INFO, appInfo);
+        return this;
+    }
+
+    /**
+    * @return {AppInfo}
+    */
+    getAppInfo() {
+        return this.getParameter(RegisterAppInterface.KEY_APP_INFO);
+    }
+
+    /**
+     * @param {TemplateColorScheme} dayColorScheme
+     * @return {RegisterAppInterface}
+     */
+    setDayColorScheme(dayColorScheme) {
+        this.validateType(TemplateColorScheme, dayColorScheme);
+
+        this.setParameter(RegisterAppInterface.KEY_DAY_COLOR_SCHEME, dayColorScheme);
+        return this;
+    }
+
+    /**
+    * @return {TemplateColorScheme}
+    */
+    getDayColorScheme() {
+        return this.getParameter(RegisterAppInterface.KEY_DAY_COLOR_SCHEME);
+    }
+
+    /**
+     * @param {TemplateColorScheme} nightColorScheme
+     * @return {RegisterAppInterface}
+     */
+    setNightColorScheme(nightColorScheme) {
+        this.validateType(TemplateColorScheme, nightColorScheme);
+
+        this.setParameter(RegisterAppInterface.KEY_NIGHT_COLOR_SCHEME, nightColorScheme);
+        return this;
+    }
+
+    /**
+    * @return {TemplateColorScheme}
+    */
+    getNightColorScheme() {
+        return this.getParameter(RegisterAppInterface.KEY_NIGHT_COLOR_SCHEME);
+    }
+}
+
+RegisterAppInterface.KEY_SDL_MSG_VERSION = 'syncMsgVersion';
+RegisterAppInterface.KEY_APP_NAME = 'appName';
+RegisterAppInterface.KEY_TTS_NAME = 'ttsName';
+RegisterAppInterface.KEY_NGN_MEDIA_SCREEN_APP_NAME = 'ngnMediaScreenAppName';
+RegisterAppInterface.KEY_VR_SYNONYMS = 'vrSynonyms';
+RegisterAppInterface.KEY_IS_MEDIA_APPLICATION = 'isMediaApplication';
+RegisterAppInterface.KEY_LANGUAGE_DESIRED = "languageDesired";
+RegisterAppInterface.KEY_HMI_DISPLAY_LANGUAGE_DESIRED = "hmiDisplayLanguageDesired";
+RegisterAppInterface.KEY_APP_HMI_TYPE = "appHMIType";
+RegisterAppInterface.KEY_HASH_ID = "hashID";
+RegisterAppInterface.KEY_DEVICE_INFO = "deviceInfo";
+RegisterAppInterface.KEY_APP_ID = "appID";
+RegisterAppInterface.KEY_FULL_APP_ID = "fullAppID";
+RegisterAppInterface.KEY_APP_INFO = "appInfo";
+RegisterAppInterface.KEY_DAY_COLOR_SCHEME = "dayColorScheme";
+RegisterAppInterface.KEY_NIGHT_COLOR_SCHEME = "nightColorScheme";
+RegisterAppInterface.APP_ID_MAX_LENGTH = 10;
```

Comment:
from where came this parameter? I can't find equivalent in `MOBILE_API.xml`

Reply:
See previous comment about corner cases. This was detailed in a proposal as well.